INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 09:53:53] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 09:53:54] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 09:53:54] "[33mGET /favicon.ico HTTP/1.1[0m" 404 -
ERROR:app:Exception on /agregar [GET]
Traceback (most recent call last):
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\routes\personas\personas.py", line 24, in agregar
    return render_template('agregar.html',form=personaForm)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\templating.py", line 152, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\templating.py", line 133, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\routes\personas\templates\agregar.html", line 13, in top-level template code
    {{forma.csrf_token}}
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\jinja2\environment.py", line 485, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'forma' is undefined
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 09:53:58] "[35m[1mGET /agregar HTTP/1.1[0m" 500 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 09:59:04] "GET / HTTP/1.1" 200 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 09:59:24] "GET / HTTP/1.1" 200 -
ERROR:app:Exception on /agregar [GET]
Traceback (most recent call last):
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\routes\personas\personas.py", line 24, in agregar
    return render_template('agregar.html',form=personaForm)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\templating.py", line 152, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\templating.py", line 133, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\routes\personas\templates\agregar.html", line 13, in top-level template code
    {{forma.csrf_token}}
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\jinja2\environment.py", line 485, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'forma' is undefined
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 09:59:32] "[35m[1mGET /agregar HTTP/1.1[0m" 500 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:02:48] "GET / HTTP/1.1" 200 -
ERROR:app:Exception on /agregar [GET]
Traceback (most recent call last):
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\routes\personas\personas.py", line 24, in agregar
    return render_template('agregar.html',form=personaForm)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\templating.py", line 152, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\templating.py", line 133, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\routes\personas\templates\agregar.html", line 13, in top-level template code
    {{forma.csrf_token}}
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\jinja2\environment.py", line 485, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'forma' is undefined
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:02:50] "[35m[1mGET /agregar HTTP/1.1[0m" 500 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:04:17] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:04:19] "GET /agregar HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:05:06] "[32mPOST /agregar HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:05:06] "GET /index HTTP/1.1" 200 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:08:35] "GET /index HTTP/1.1" 200 -
ERROR:app:Exception on /editar/1 [GET]
Traceback (most recent call last):
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\routes\personas\personas.py", line 35, in editar
    return render_template('editar.html',forma=personaForm)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\templating.py", line 151, in render_template
    template = app.jinja_env.get_or_select_template(template_name_or_list)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\jinja2\environment.py", line 1081, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\jinja2\environment.py", line 1010, in get_template
    return self._load_template(name, globals)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\jinja2\environment.py", line 969, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\jinja2\loaders.py", line 126, in load
    source, filename, uptodate = self.get_source(environment, name)
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\templating.py", line 65, in get_source
    return self._get_source_fast(environment, template)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\templating.py", line 99, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: editar.html
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:08:37] "[35m[1mGET /editar/1 HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:09:29] "[32mGET /eliminar/1 HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:09:29] "GET /index HTTP/1.1" 200 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:22:05] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:22:05] "[33mGET /favicon.ico HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:22:06] "GET /agregar HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:22:37] "[32mPOST /agregar HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:22:37] "GET /index HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:22:42] "GET /editar/2 HTTP/1.1" 200 -
ERROR:app:Exception on /editar/2 [POST]
Traceback (most recent call last):
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Cesar Castro\Documents\Universidad\Phython\Flask2\routes\personas\personas.py", line 32, in editar
    personaForm.populate_object(persona)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'PersonaForm' object has no attribute 'populate_object'
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:22:56] "[35m[1mPOST /editar/2 HTTP/1.1[0m" 500 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:24:13] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:24:42] "GET /editar/2 HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:24:46] "[32mPOST /editar/2 HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [18/Oct/2023 10:24:46] "GET /index HTTP/1.1" 200 -
